pdfminer/pdfdocument.py

```diff
@@ -895,6 +901,105 @@ def search(entry: object, level: int
+                objs = list_value(td['Nums'])
+                for (k, v) in choplist(2, objs):
+                    yield int_value(k), dict_value(v)
+
+            if 'Kids' in td:  # Intermediate node
+                for child_ref in list_value(td['Kids']):
+                    yield from walk_number_tree(dict_value(child_ref))
+
+        # Extract and sanity-check ranges
+        ranges = list(walk_number_tree(labels_tree))
+
+        # The tree should be sorted
+        if settings.STRICT:
+            if not all(a[0] <= b[0] for a, b in zip(ranges, ranges[1:])):
+                raise PDFSyntaxError('PageLabels are out of order')
+        else:
+            ranges.sort(key=lambda t: t[0])
+
+        # The tree must begin with page index 0
+        if len(ranges) == 0 or ranges[0][0] != 0:
+            if settings.STRICT:
+                raise PDFSyntaxError('PageLabels is missing page index 0')
+
+            # Try to cope, by assuming empty labels for the initial pages
+            ranges.insert(0, (0, {}))
+
+        def emit_labels(
+            ranges: List[Tuple[int, Dict[Any, Any]]],
+            total_pages: Optional[int]
+        ) -> Iterator[str]:
+            """
+            Walk a list of ranges and label dicts, yielding label strings.
+            """
+            for (i, (range_start, label_dict)) in enumerate(ranges):
+                style = label_dict.get('S')
+                prefix = decode_text(str_value(label_dict.get('P', b'')))
+                first = int_value(label_dict.get('St', 1))
+
+                def mkrange(limit: int) -> range:
+                    "Construct a suitable range for the values to format."
+                    return range(first, first + limit - range_start)
+
+                if i + 1 == len(ranges):
+                    # This is the last specified range. It continues until
+                    # the end of the document, which may be unknown.
+                    if total_pages is None:
+                        values: Iterable[int] = itertools.count(first)
+                    else:
+                        values = mkrange(total_pages)
+                else:
+                    values = mkrange(ranges[i + 1][0])
+
+                for value in values:
+                    if style is LIT('D'):    # Decimal arabic numerals
+                        label = str(value)
+                    elif style is LIT('R'):  # Uppercase roman numerals
+                        label = format_int_roman(value).upper()
+                    elif style is LIT('r'):  # Lowercase roman numerals
+                        label = format_int_roman(value)
+                    elif style is LIT('A'):  # Uppercase letters A-Z, AA-ZZ...
+                        label = format_int_alpha(value).upper()
+                    elif style is LIT('a'):  # Lowercase letters a-z, aa-zz...
+                        label = format_int_alpha(value)
+                    else:
+                        label = ''
```

Comment:
Use `logger.warning` to inform that something unexpected happened.

Reply:
Or is this catching the case where there is no style specified and the label is only the prefix?

Reply:
> Or is this catching the case where there is no style specified and the label is only the prefix?

It was, but it didn't need to be this sloppy. Fixed.